presto-main/src/main/java/com/facebook/presto/sql/planner/DependencyExtractor.java

```diff
@@ -36,8 +37,13 @@ private DependencyExtractor() {}
 
     public static Set<Symbol> extractUnique(PlanNode node)
     {
+        return extractUnique(node, n -> n);
+    }
+
+    public static Set<Symbol> extractUnique(PlanNode node, Lookup lookup)
```

Comment:
> How is that possible if I didn't change the behavior here?

What has been changed is that you added handling of `GroupReferences`: "if instanceof GroupReference, lookup & recurse"